lib/src/extrema.dart

```diff
@@ -0,0 +1,96 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// extrema.dart
+// Implementation of finding extremas (max or min) of signals.
+//
+// 2024 September 16
+// Author: Roberto Torres <roberto.torres@intel.com>
+//
+
+import 'package:rohd/rohd.dart';
+import 'package:rohd_hcl/rohd_hcl.dart';
+
+/// Determines the extremas (maximum or minimum) of a List<[Logic]>.
+
+class Extrema extends Module {
+  /// The [index] of the extrema.
+  Logic get index => output('index');
+
+  /// The [val] of the extrema.
+  Logic get val => output('val');
+
+  /// Finds an extrema of List<[Logic]> [toCompare]. Inputs need not be the same
+  /// width, and will all be considered positive unsigned numbers.
+  ///
+  /// If [max] is `true`, will find maximum value, else will find minimum.
+  /// If [first] is `true`, will find first extrema, else will find last.
+  ///
+  /// Outputs the [index] and [val] of the extrema in the list of [toCompare].
+
+  Extrema(List<Logic> toCompare, {bool max = true, bool first = true}) {
```

Comment:
i think the `first` argument doesn't really make sense here, since effictively `max` and `first` are controlling the same thing (min vs max).  Addiitonal useful configurability would be something like `n` where you can find the "n'th" biggest or smallest.

Reply:
The idea here was in the case of multiple instances of the same max or min, one could choose to return the index of the first instance or the last.

Reply:
oh i see, i think it's at least confusing, but also probably not necessary.  i'd remove it